WordPress/Sniffs/WP/UseCapabilitiesNotRolesSniff.php

```diff
@@ -0,0 +1,234 @@
+		'level_9'                     => true,
+		'level_8'                     => true,
+		'level_7'                     => true,
+		'level_6'                     => true,
+		'level_5'                     => true,
+		'level_4'                     => true,
+		'level_3'                     => true,
+		'level_2'                     => true,
+		'level_1'                     => true,
+		'level_0'                     => true,
+		'create_sites'                => true,
+		'delete_sites'                => true,
+		'manage_network'              => true,
+		'manage_sites'                => true,
+		'manage_network_users'        => true,
+		'manage_network_plugins'      => true,
+		'manage_network_themes'       => true,
+		'manage_network_options'      => true,
+		'delete_site'                 => true,
+		'upgrade_network'             => true,
+		'setup_network'               => true,
+		'upload_plugins'              => true,
+		'upload_themes'               => true,
+		'customize'                   => true,
+		'add_users'                   => true,
+		'install_languages'           => true,
+		'update_languages'            => true,
+		'deactivate_plugins'          => true,
+		'upgrade_php'                 => true,
+		'export_others_personal_data' => true,
+		'erase_others_personal_data'  => true,
+		'manage_privacy_options'      => true,
+		'edit_categories'             => true,
+		'delete_categories'           => true,
+		'manage_post_tags'            => true,
+		'edit_post_tags'              => true,
+		'delete_post_tags'            => true,
+		'edit_css'                    => true,
+		'assign_categories'           => true,
+		'assign_post_tags'            => true,
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+
+		$position = $this->target_functions[ $matched_content ];
+		if ( ! isset( $parameters[ $position ] ) ) {
+			return;
+		}
+
+		$matched_parameter = $this->strip_quotes( $parameters[ $position ]['raw'] );
+		if ( isset( $this->core_capabilities[ $matched_parameter ] ) ) {
+			return;
+		}
+
+		if ( isset( $this->core_roles[ $matched_parameter ] ) ) {
+			$this->phpcsFile->addError(
+				'Capabilities should be used instead of roles. Found "%s" in function "%s"',
+				$stackPtr,
+				'RoleFound',
+				array(
+					$matched_parameter,
+					$matched_content,
+				)
+			);
+		} else {
+			$this->phpcsFile->addWarning(
+				'"%s" is an unknown role or capability. Check the "%s()" function call to ensure it is a capability and not a role.',
```

Comment:
Be aware that this will give a lot of false positives/unnecessary warnings as the `$capability` a plugin needs is often defined in a class property/(class) constant and passed to the function that way.

As I can imagine that this check is useful for the Theme Review Team, maybe have it disabled by default with the ability to enable it via a custom property ?

```php
class My_Plugin {
    const CAPABILITY = 'manage_options';

    public function do_something() {
        if ( current_user_can( self::CAPABILITY ) ) {}
    }
}
```